auth-foundation/src/main/java/com/okta/authfoundation/credential/Token.kt

```diff
@@ -114,6 +116,12 @@ class Token(
             issuedTokenType,
         )
     }
+
+    data class Metadata(
+        val id: String,
+        val tags: Map<String, String>,
+        val payloadData: JsonObject
```

Comment:
The Swift `Token.Metadata` class conforms to the [HasClaims](https://okta.github.io/okta-mobile-swift/development/authfoundation/documentation/authfoundation/token/metadata/hasclaims-implementations) and [ClaimsContainer](https://okta.github.io/okta-mobile-swift/development/authfoundation/documentation/authfoundation/token/metadata/claimcontainer-implementations) protocols, which enable the use of conveniences to interact with the set of claims contained in it.  This enables the developer to interact with the metadata like so:

```
metadata.address.contains("some substring")
metadata.subject == "some_user_id"
metadata.customClaims.filter { $0 == "someCustomClaim" }
```

And so on. Those protocols are used on the [JWT](https://okta.github.io/okta-mobile-swift/development/authfoundation/documentation/authfoundation/jwt) class as well, enabling consistency in how these values are accessed between JWTs and token metadata.

Reply:
Good call. Yeah, there is a ClaimsProvider class that allows doing this. I will add that to Metadata for easier processing of claims.